### PR TITLE
Repository Management

### PR DESCRIPTION
**Pull request by dhoehna**
## Summary of the pull request
The main contribution is database migration and the tests for database migration.

The second contribution is removing the repository section from File Explorer Customization.

Smaller changes

1. Implemented some edge cases for repository actions.
    1.  Moving repositories across drives.
4. Added a progress ring when moving repositories.
5. Deleting a repository is now behind the scenes.
6. Moved Repository Management out of experimental.
7. Moved Repository Management to be under "Machine Configuration" in the nav menu.
8. Removing the old repository location after moving.

## References and relevant issues

I will make another PR with documentation on everything, including how to migrate the database.

## Validation steps performed

Movies and tests.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

Movies!
Database migration in action!
DatabaseMigration

Cloning Repositories into Repository Manager!
CloningRepositoriesIntoRepositoryManager

Adding an existing repository!
AddingAnExistingRepository

Repository actions except Move, Hide, and Delete!
https://github.com/user-attachments/assets/dad1899f-2c05-4881-9648-261f97a6ca77

Move, Hide, and Delete!
https://github.com/user-attachments/assets/1f8aae5e-b1bf-490c-9bbb-e0326f8c9eac

### CODE REVIEW

File: tools/Customization/DevHome.Customization/Views/FileExplorerPage.xaml [hunk 1]
```diff
@@ -13,7 +13,6 @@
             <StackPanel MaxWidth="{ThemeResource MaxPageContentWidth}" Margin="{ThemeResource ContentPageMargin}">
                 <views:FileExplorerView />
                 <views:VersionControlIntegrationSettingsView Visibility="{x:Bind ViewModel.IsFeatureEnabled}"/>
-                <views:AddRepositoriesView Visibility="{x:Bind ViewModel.IsFeatureEnabled}"/>
```
Comment (ssparach) [inline tools/Customization/DevHome.Customization/Views/FileExplorerPage.xaml:16]: The AddRepositoriesView may need to be present for a while inside Dev Home and cannot be removed. Starting an offline email thread for further discussion. Thanks!

**Review by krschau — CHANGES_REQUESTED** [review]
Holding check-ins for the moment.

**Comment by dhoehna** [comment]
Moving into draft because the team wants to withhold any feature PR's until a later date.
